### ISSUE TITLE
[Question] txt or pdf file not permitted?

### ISSUE THREAD

**Issue by scheung38** [uzochukwueddie/react-dropzone#3]
How to allow text documents to be sent?

**Comment by uzoeddie** [comment]
Do you mean a file with a .txt extension? if yes, you can add it to the validTypes array

**Comment by scheung38** [comment]
At the moment .txt not permitted, so validTypes when added for .txt .pdf .xlsx I can upload anything. Would having this as an UI option allow different end users to customiz a good feature? 

But on the same token you mentioned imgBB only accepts images?

**Comment by scheung38** [comment]
![Screenshot 2020-09-04 at 15 39 14](https://user-images.githubusercontent.com/1273014/92251456-cbe4ed80-eec4-11ea-955c-2e75e3852a75.png)
 like this?

**Comment by scheung38** [comment]
![Screenshot 2020-09-04 at 15 39 59](https://user-images.githubusercontent.com/1273014/92251541-e61ecb80-eec4-11ea-89fc-2fa2cdf5f372.png)

**Comment by uzoeddie** [comment]
txt is not an image type. try using this text/plain instead image/txt

**Comment by uzoeddie** [comment]
checkout different MIME types
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

**Comment by scheung38** [comment]
'text/plain' seems to accept in the app, but not uploadable:

![Screenshot 2020-09-04 at 15 44 12](https://user-images.githubusercontent.com/1273014/92251948-7ceb8800-eec5-11ea-909e-6bfbbd972558.png)


![Screenshot 2020-09-04 at 15 44 24](https://user-images.githubusercontent.com/1273014/92251970-84129600-eec5-11ea-85d8-891b4b1f988f.png)

'text/*' is not even permitted, so this means we still cannot upload tex documents? pdf?

'application/pdf'

if changing to another cloud storage like Azure blob storage then it is just a single line of change?

**Comment by uzoeddie** [comment]
Yes, it won't upload because the cloud storage used in the tutorial only accepts images. If you have another cloud storage that accepts text files, you can try that

**Comment by scheung38** [comment]
'application/pdf' is permitted and uploaded!

**Comment by scheung38** [comment]
like to know how to close modal window with ESC instead of having to move with mouse click, better user experience thanks

**Comment by uzoeddie** [comment]
Try this. I have not tried it though

```
const escapeKey (event) => {
    if (event.key === 'Escape') {
      closeModal()
    }
 }
```

Then add 
`window.addEventListener('keydown', escapeKey);`
into the useEffect hook

**Comment by scheung38** [comment]
In the case of .txt and .pdf both selected:

![Screenshot 2020-09-04 at 16 42 33](https://user-images.githubusercontent.com/1273014/92257748-a5778000-eecd-11ea-984d-6a70d56c8715.png)

a bit confusing, yes pdf is Uploaded and yes red to indicate .txt not valid...but this is showing both for users.

**Comment by scheung38** [comment]
add the above in Dropzone.js correct?

**Comment by scheung38** [comment]
if placed inside useEffect:

![Screenshot 2020-09-04 at 16 56 34](https://user-images.githubusercontent.com/1273014/92260297-9a255400-eecf-11ea-8649-9bad7d877e83.png)

**Comment by uzoeddie** [comment]
Did you try moving the escape function above the useEffect?
